client/src/js/services/journal/journalColumnConfig.js

```diff
@@ -0,0 +1,130 @@
+angular.module('bhima.services')
+.service('JournalColumnConfigService', JournalColumnConfigService);
+
+JournalColumnConfigService.$inject = ['uiGridConstants', 'AppCache'];
+
+/**
+ * Posting Journal Column config Service 
+ * This service is responsible of Showing and Hiding column
+*/
+function JournalColumnConfigService(uiGridConstants, AppCache) { 
+  var service = this; 
+  
+  // Variable used to track and share the current grids API object
+  var gridApi;
+
+  //An appcache instance
+  //@todo use ui-grid-saveState
+  var cache = AppCache('columnConfigService');
+
+  //default visibility configuartion
+  var defaultVisibility = [ 
```

Comment:
I began with a simple JSON exaclty like that, but I have noticed that I need to display column name   on the modal to configure visibility.